packages/internals/src/get-generators/generatorResolvers/prisma-client-js/prismaClientResolver.ts

```diff
@@ -0,0 +1,115 @@
+import Debug from '@prisma/debug'
+import chalk from 'chalk'
+import fs from 'fs'
+import path from 'path'
+
+import { findPrismaClientDir } from './auto-installation/findPrismaClientDir'
+import { getPackageCmd } from './auto-installation/getPackageCmd'
+import { runPackageCmd } from './auto-installation/runPackageCmd'
+import { checkTypeScriptVersion } from './check-dependencies/checkTypeScriptVersion'
+import { checkYarnVersion } from './check-dependencies/checkYarnVersion'
+import { isYarnUsed } from './check-dependencies/isYarnUsed'
+import { resolvePkg } from './check-dependencies/resolve'
+
+export const debug = Debug('prisma:generator')
+
+/**
+ * Client generator resolver. The generator is shipped with the Client, so if
+ * the client is not installed, it will be installed unless generation is
+ * skipped.
+ * @param baseDir
+ * @param version
+ * @returns
+ */
+export async function prismaClientResolver(baseDir: string, version?: string) {
+  let prismaClientDir = await findPrismaClientDir(baseDir)
+
+  debug('baseDir', baseDir)
+
+  checkYarnVersion()
+  await checkTypeScriptVersion()
+
+  if (!prismaClientDir && !process.env.PRISMA_GENERATE_SKIP_AUTOINSTALL) {
+    // TODO: should this be relative to `baseDir` rather than `process.cwd()`?
+    if (
+      !fs.existsSync(path.join(process.cwd(), 'package.json')) &&
+      !fs.existsSync(path.join(process.cwd(), '../package.json'))
+    ) {
+      // Create default package.json
+      const defaultPackageJson = `{
```

Comment:
Do you know why are we creating `package.json` outselves in this case?

Reply:
I think: `npm add` below will otherwise traverse up the tree to find a `package.json` where to put the dependencies - which might be outside of the project, leading to very funky behavior.

Reply:
I haven't touched the package.json creation part which is also likely to be problematic too. But to answer your question, we also create a package.json for the user to init their project and for the reasons Jan pointed out. For reference, this is the file that we write, that already smells like an issue:

```sh
{"name":"my-prisma-project","version":"1.0.0","description":"","main":"index.js","scripts":{"test":"echo \"Error: no test specified\" && exit 1"},"keywords":[],"author":"","license":"ISC","devDependencies":{"prisma":"^4.13.0-integration-fix-use-ni-pm-detection.15"},"dependencies":{"@prisma/client":"^4.13.0-integration-fix-use-ni-pm-detection.15"}}
```

Reply:
I haven't touched the package.json creation part which is also likely to be problematic too. But to answer your question, we also create a package.json for the user to init their project and for the reasons Jan pointed out. For reference, this is the file that we write, that already smells like an issue:

```sh
{"name":"my-prisma-project","version":"1.0.0","description":"","main":"index.js","scripts":{"test":"echo \"Error: no test specified\" && exit 1"},"keywords":[],"author":"","license":"ISC","devDependencies":{"prisma":"^4.13.0-integration-fix-use-ni-pm-detection.15"},"dependencies":{"@prisma/client":"^4.13.0-integration-fix-use-ni-pm-detection.15"}}
```